task3/src/routes/user/index.js

```diff
@@ -1,65 +1,87 @@
 import express from 'express';
 
-import {
-    getUserById,
-    updateUser,
-    deleteUser,
-    createUser
-} from '../../usersData';
 import validators from './validators';
+import UserService from '../../services/userService';
+import { UserModel } from '../../models';
 
+const userService = new UserService(UserModel);
 const router = express.Router();
 
 router
     .route('/:id')
-    .get((req, res) => {
-        const { id: userId } = req.params;
-        const user = getUserById(userId);
-        if (!user) {
-            res.status(404).json({
-                message: `User with id ${userId} not found`
+    .get(validators.uuid, async (req, res) => {
+        try {
+            const { id: userId } = req.params;
+            const user = await userService.getById(userId);
+            if (!user) {
+                res.status(404).json({
+                    message: `User with id ${userId} not found`
+                });
+            } else {
+                res.json(user);
+            }
+        } catch (err) {
+            res.status(err.status || 500).json({
+                message: err
             });
-        } else {
-            res.json(user);
         }
     })
-    .delete((req, res) => {
-        const { id: userId } = req.params;
-        const user = getUserById(userId);
-        if (!user) {
-            res.status(404).json({
-                message: `User with id ${userId} not found`
-            });
-        } else {
-            deleteUser(userId);
-            res.json({
-                message: `User with id ${userId} deleted`
+    .delete(validators.uuid, async (req, res) => {
+        try {
+            const { id: userId } = req.params;
+            const user = await userService.getById(userId);
```

Comment:
Is it OK that delete and put controllers call userService.getById ?

Reply:
Why not? What potential problems do you see here? Services are created to encapsulate data manipulation logic

Reply:
Ok 👍